polaris-react/src/components/Button/Button.tsx

```diff
@@ -242,12 +263,10 @@ export function Button({
   return buttonMarkup;
 }
 
-function isIconSource(x: any): x is IconSource {
-  return (
-    typeof x === 'string' ||
-    (typeof x === 'object' && x.body) ||
-    typeof x === 'function'
-  );
+function isIconSourceLike(
+  x?: React.ReactElement | IconSource,
+): x is IconSource {
+  return isValidElement(x);
 }
```

Comment:
Second pair of eyes on this would be great. Not sure what case `x === object && x.body` was handling, however, following the control flow `isValidElement` seemed like the minimum requirement.

Reply:
I did some digging and it looks like legacy `Icon` behavior that wasn't cleaned up.
https://github.com/Shopify/polaris/pull/926/files#diff-08acd9c461dd2b3858a54d0dd361b9164bdc650ee0050ace49c108d9dc6e2059L220-L226

I ended up removing this utility and inline the behavior.